packages/wonder-stuff-testing/src/jest/__tests__/after-each-restore-env.test.ts

```diff
@@ -0,0 +1,146 @@
+import * as JestWrappers from "../internal/jest-wrappers";
+
+import {afterEachRestoreEnv} from "../after-each-restore-env";
+
+jest.mock("../internal/jest-wrappers");
+
+describe("#afterEachRestoreEnv", () => {
+    const EXISTS_1 = process.env.EXISTS_1;
+    const EXISTS_2 = process.env.EXISTS_2;
+
+    const ABSENT_1 = process.env.ABSENT_1;
+    const ABSENT_2 = process.env.ABSENT_2;
+
+    afterEach(() => {
+        // In case our tests misbehave, we still want to be good test denizens
+        // so we restore the environment variables we changed without using the
+        // code under test.
+        if (EXISTS_1 === undefined) {
+            delete process.env.EXISTS_1;
+        } else {
+            process.env.EXISTS_1 = EXISTS_1;
+        }
+
+        if (EXISTS_2 === undefined) {
+            delete process.env.EXISTS_2;
+        } else {
+            process.env.EXISTS_2 = EXISTS_2;
+        }
+
+        if (ABSENT_1 === undefined) {
+            delete process.env.ABSENT_1;
+        } else {
+            process.env.ABSENT_1 = ABSENT_1;
+        }
+
+        if (ABSENT_2 === undefined) {
+            delete process.env.ABSENT_2;
+        } else {
+            process.env.ABSENT_2 = ABSENT_2;
+        }
+    });
```

Comment:
**note:** This serves as a good example of why this is a useful thing to have. This whole thing could be replaced with:

```ts
afterEachRestoreEnv("EXPECT_1", "EXPECT_2", "ABSENT_1", "ABSENT_2");

// or, if we don't need to be exact:

afterEachRestoreEnv();
```